### ISSUE TITLE
Contract checking does not work for @Contract("null -> null")

### ISSUE THREAD

**Issue by agrieve** [uber/NullAway#1121]
I'd like to say that if the given method returns non-null, then the passed in value is non-null.

```java
    @Contract("null -> null")
    public static @Nullable Activity activityFromContext(@Nullable Context context) {
        // Only retrieves the base context if the supplied context is a ContextWrapper but not an
        // Activity, because Activity is a subclass of ContextWrapper.
        while (context instanceof ContextWrapper) {
            if (context instanceof Activity) return (Activity) context;

            context = ((ContextWrapper) context).getBaseContext();
        }

        return null;
    }
```

But this still gives a warning:
```
        if (ContextUtils.activityFromContext(context) != null) {
            mDialogCore = new DialogOverlayCore();
            // Complains that context is nullable:
            mDialogCore.initialize(context, mConfig, DialogOverlayImpl.this, mAsPanel);
            mDialogCore.onWindowToken(window.getWindowToken());
        }
```

I suspect this might fall under:

> [Not all possible clauses of @Contract annotations are fully parsed or supported by NullAway](https://github.com/uber/NullAway/wiki/Supported-Annotations#contracts:~:text=Not%20all%20possible%20clauses%20of%20%40Contract%20annotations%20are%20fully%20parsed%20or%20supported%20by%20NullAway)

If so, it'd be great to update the docs to say that contract support cannot infer parameters based on nullness of return types.

**Comment by msridhar** [comment]
Interesting!  I feel like we should be able to handle this one, but I'll need to dig into the code more to see if there is some fundamental reason this is hard.

**Comment by msridhar** [comment]
I think you're right, actually, that handling this case and inferring parameter nullness from the return nullness is kind of beyond what our code is set up to do right now.  I'll think more, but probably we'll need to just update the docs.

**Comment by msridhar** [comment]
Documentation has been updated; thanks for this report.
